internal/controllers/configuration/kongupstreampolicy_utils.go

```diff
@@ -157,6 +150,52 @@ func (r *KongUpstreamPolicyReconciler) buildServicesStatus(ctx context.Context,
 	return servicesStatus, nil
 }
 
+// httpRouteHasUpstreamPolicyConflictedBackendRefsWithService checks if there's any HTTPRoute's rule that uses multiple backendRefs
+// AND they're not all using the same KongUpstreamPolicy.
+// If so, that means that we have a conflict because we cannot apply multiple KongUpstreamPolicy to the same Kong Service.
+func httpRouteHasUpstreamPolicyConflictedBackendRefsWithService(
+	httpRoute gatewayapi.HTTPRoute,
+	upstreamPolicyServices map[string]indexedServiceStatus,
+	serviceKey string,
+) bool {
+	backendRefsUsedWithThisService := getAllBackendRefsUsedWithService(httpRoute, serviceKey)
+	_, hasAnyBackendRefNotUsingSameUpstreamPolicy := lo.Find(backendRefsUsedWithThisService, func(br gatewayapi.HTTPBackendRef) bool {
+		serviceRef := backendRefToServiceRef(httpRoute.Namespace, br.BackendRef)
+		if serviceRef == "" {
+			return false
+		}
+		// If the serviceRef is not in the upstreamPolicyServices, it means it doesn't use this KongUpstreamPolicy.
+		_, ok := upstreamPolicyServices[serviceRef]
+		return !ok
+	})
+	return hasAnyBackendRefNotUsingSameUpstreamPolicy
+}
+
+// getAllBackendRefsUsedWithService returns HTTPRoute's backendRefs that use the given service (excluding the given service).
+func getAllBackendRefsUsedWithService(httpRoute gatewayapi.HTTPRoute, serviceKey string) []gatewayapi.HTTPBackendRef {
+	var backendRefs []gatewayapi.HTTPBackendRef
+	for _, rule := range httpRoute.Spec.Rules {
+		rule := rule
```

Comment:
Why do we add an assignment here? I did not see `rule` modified in the loop body or concurrency issues.

Reply:
It's because I couldn't make `looppointer` to respect `//nolint` directives. Please take a look at d8b15ee where I found a solution to do that (I had to force `nolintlint` to not lint the directive and change it from `// nolint` to `//nolint` because that would be not respected by `looppointer` 🤣).